test/e2e/upgrades/apps/deployments.go

```diff
@@ -140,47 +134,9 @@ func (t *DeploymentUpgradeTest) Test(f *framework.Framework, done <-chan struct{
 
 	By(fmt.Sprintf("Waiting for deployment %q to complete adoption", deployment.Name))
 	framework.ExpectNoError(framework.WaitForDeploymentStatus(c, deployment))
-
-	By(fmt.Sprintf("Checking that replica sets for deployment %q are the same as prior to the upgrade", t.updatedD.Name))
-	_, allOldRSs, newRS, err := deploymentutil.GetAllReplicaSets(t.updatedD, c.ExtensionsV1beta1())
-	framework.ExpectNoError(err)
-	if newRS == nil {
-		By(t.spewDeploymentAndReplicaSets(newRS, allOldRSs))
-		framework.ExpectNoError(fmt.Errorf("expected a new replica set for deployment %q", t.updatedD.Name))
-	}
-	if newRS.UID != t.newRS.UID {
-		By(t.spewDeploymentAndReplicaSets(newRS, allOldRSs))
-		framework.ExpectNoError(fmt.Errorf("expected new replica set:\n%#v\ngot new replica set:\n%#v\n", t.newRS, newRS))
-	}
-	if len(allOldRSs) != 1 {
-		By(t.spewDeploymentAndReplicaSets(newRS, allOldRSs))
-		errString := fmt.Sprintf("expected one old replica set, got %d\n", len(allOldRSs))
-		for i := range allOldRSs {
-			rs := allOldRSs[i]
-			errString += fmt.Sprintf("%#v\n", rs)
-		}
-		framework.ExpectNoError(fmt.Errorf(errString))
-	}
-	if allOldRSs[0].UID != t.oldRS.UID {
```

Comment:
You can remove the check that compares RSes before and after the upgrade, like this line. But leave other checks as is to make sure Deployment history is still intact.